omegaconf/built_in_resolvers.py

```diff
@@ -0,0 +1,167 @@
+import os
+import warnings
+
+# from collections.abc import Mapping, MutableMapping
+from typing import Any, List, Mapping, Optional, Union
+
+from ._utils import _DEFAULT_MARKER_, Marker, _get_value, decode_primitive
+from .base import Container
+from .basecontainer import BaseContainer
+from .dictconfig import DictConfig
+from .errors import ConfigKeyError, ValidationError
+from .grammar_parser import parse
+from .listconfig import ListConfig
+from .nodes import AnyNode
+from .omegaconf import OmegaConf
+
+# Special marker use as default value when calling `OmegaConf.select()`. It must be
+# different from `_DEFAULT_MARKER_`, which is used by `OmegaConf.select()`.
+_DEFAULT_SELECT_MARKER_: Any = Marker("_DEFAULT_SELECT_MARKER_")
+
+
+def decode(expr: Optional[str], _parent_: Container) -> Any:
+    """
+    Parse and evaluate `expr` according to the `singleElement` rule of the grammar.
+
+    If `expr` is `None`, then return `None`.
+    """
+    if expr is None:
+        return None
+
+    if not isinstance(expr, str):
+        raise TypeError(
+            f"`oc.decode` can only take strings or None as input, "
+            f"but `{expr}` is of type {type(expr).__name__}"
+        )
+
+    parse_tree = parse(expr, parser_rule="singleElement", lexer_mode="VALUE_MODE")
+    val = _parent_.resolve_parse_tree(parse_tree)
+    return _get_value(val)
+
+
+def dict_keys(
+    in_dict: Union[str, Mapping[Any, Any]],
+    _root_: BaseContainer,
+    _parent_: Container,
+) -> ListConfig:
+    in_dict = _get_and_validate_dict_input(
+        in_dict, root=_root_, resolver_name="oc.dict.keys"
+    )
+    assert isinstance(_parent_, BaseContainer)
+
+    ret = OmegaConf.create(list(in_dict.keys()), parent=_parent_)
+    assert isinstance(ret, ListConfig)
+    return ret
+
+
+def dict_values(
+    in_dict: Union[str, Mapping[Any, Any]], _root_: BaseContainer, _parent_: Container
+) -> ListConfig:
+    in_dict = _get_and_validate_dict_input(
+        in_dict, root=_root_, resolver_name="oc.dict.values"
+    )
+
+    if isinstance(in_dict, DictConfig):
```

Comment:
We can. It would prevent it from being used on the output of a resolver that's a dictionary though (instead requiring the somewhat cumbersome and less efficient workaround of creating a temporary node to store the dict as a DictConfig).
I don't feel strongly about it though (it's a niche use case) so let me know if you prefer to support only DictConfigs.

Reply:
Okay, let's simplify then.
We can always add support if there is enough demand, but it's much harder to take support away.